lib/chef/config.rb

```diff
@@ -49,5 +61,24 @@ class Config
       evt_loggers
     end
 
+    # Override the default values that were set by Ohai.
+    #
+    # REMOVEME once these configurables are removed from the top level of Ohai.
+    default :log_level, LOG_LEVEL
+    default :log_location, LOG_LOCATION
+
+    # Ohai::Config[:log_level] is deprecated and warns when set. Unfortunately,
+    # there is no way to distinguish between setting log_level and setting
+    # Ohai::Config[:log_level]. Since log_level and log_location are used by
+    # chef-client and other tools (e.g., knife), we will mute the warnings here
+    # by redefining the config_attr_writer to not warn for these options.
+    #
+    # REMOVEME once the warnings for these configurables are removed from Ohai.
```

Comment:
This seems similar to the issue dealt with in https://github.com/chef/chef/blob/master/lib/chef/knife/data_bag_secret_options.rb - am I thinking about this correctly?